### PR TITLE
Enforce correct lock order between EntityTree and EntitySimulation

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

File: libraries/entities/src/SimpleEntitySimulation.cpp [hunk 1]
```diff
@@ -115,6 +115,18 @@ void SimpleEntitySimulation::clearEntitiesInternal() {
     _entitiesThatNeedSimulationOwner.clear();
 }
 
+void SimpleEntitySimulation::executeTreeOperations(EntityTree& tree) {
+    EntitySimulation::executeTreeOperations(tree);
+
+    tree.withReadLock([&] {
```
[the remark this answers is not in the document]
Comment (Atlante45) [reply libraries/entities/src/SimpleEntitySimulation.cpp]: Correct, write locks are only for when the tree structure changes or an entity moves.
Here we just mark some cells.

**Comment by Atlante45** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9119/app_12961_PR_d82333534c0cc2b303357bad68fe28aea5fdb950.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9119/HighFidelity-Beta-PR12961-d82333534c0cc2b303357bad68fe28aea5fdb950.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9119/HighFidelity-Beta-Interface-PR12961-d82333534c0cc2b303357bad68fe28aea5fdb950.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9119/HighFidelity-Beta-PR12961-d82333534c0cc2b303357bad68fe28aea5fdb950.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9119/HighFidelity-Beta-Interface-PR12961-d82333534c0cc2b303357bad68fe28aea5fdb950.dmg

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Entities are synced up and look good.

**Comment by conklin94122** [comment]
Merging per Seth.

**Pull request merged**